File: apps/desktop/src-tauri/src/core/settings/bot/providers/lifecycle/events.rs
// apps/desktop/src-tauri/src/core/settings/bot/providers/lifecycle/events.rs
// 外部依赖
use tauri::{AppHandle, Emitter};

// 内部引用
use super::resolver;
use crate::core::models::provider::check::{
    ProviderCheckCompletedPayload, ProviderCheckFailedPayload, ProviderCheckStartedPayload,
    ProviderCheckTrigger, ProviderStatusPayload,
};
use crate::core::models::provider::error::{ProviderError, ProviderIssue};
use crate::core::models::provider::ProviderId;
use crate::core::models::settings::{HealthCheckResponse, ProviderRecord};

/// 推送生命周期 started 事件
pub(super) fn emit_check_started(
    app: &AppHandle,
    run_id: &str,
    trigger: ProviderCheckTrigger,
) -> Result<(), ProviderError> {
    let payload = ProviderCheckStartedPayload {
        run_id: run_id.to_string(),
        trigger,
    };

    app.emit("providers-check-lifecycle-started", &payload)
        .map_err(|e| {
            ProviderError::LifecycleEventEmit(format!(
                "emit providers-check-lifecycle-started failed: {}",
                e
            ))
        })
}

/// 推送单个 Provider 的状态事件
pub(super) fn emit_provider_status(
    app: &AppHandle,
    run_id: &str,
    provider_id: ProviderId,
    config: ProviderRecord,
    health: HealthCheckResponse,
) -> Result<(), ProviderError> {
    let payload = ProviderStatusPayload {
        run_id: run_id.to_string(),
        provider: provider_id,
        config,
        health,
        secret_meta: resolver::resolve_provider_secret_meta(provider_id),
    };

    app.emit("provider-status", &payload).map_err(|e| {
        ProviderError::LifecycleEventEmit(format!("emit provider-status failed: {}", e))
    })
}

/// 推送生命周期 completed 事件
pub(super) fn emit_check_completed(
    app: &AppHandle,
    run_id: &str,
    failed: usize,
) -> Result<(), ProviderError> {
    let payload = ProviderCheckCompletedPayload {
        run_id: run_id.to_string(),
        failed,
    };

    app.emit("providers-check-lifecycle-completed", &payload)
        .map_err(|e| {
            ProviderError::LifecycleEventEmit(format!(
                "emit providers-check-lifecycle-completed failed: {}",
                e
            ))
        })
}

/// 推送生命周期 failed 事件
pub(super) fn emit_check_failed(
    app: &AppHandle,
    run_id: &str,
    error: &ProviderError,
    issues: Option<&[ProviderIssue]>,
) -> Result<(), ProviderError> {
    let payload = ProviderCheckFailedPayload {
        run_id: run_id.to_string(),
        code: error.code(),
        message: error.message(),
        issues,
    };

    app.emit("providers-check-lifecycle-failed", &payload)
        .map_err(|e| {
            ProviderError::LifecycleEventEmit(format!(
                "emit providers-check-lifecycle-failed failed: {}",
                e
            ))
        })
}

File: apps/ui/src/features/bot/types/provider/contract/events.ts
// apps/ui/src/features/bot/types/provider/contract/events.ts
// 内部引用
import type { ProviderId } from "../common";
import type { ProviderSecretMeta } from "./secret";
import type { ProviderRecord } from "./shared";
import type { HealthCheckResponse } from "./commands";

// ---------------------------------------------------------------------------
// Provider Check Lifecycle — 生命周期事件推送相关类型
// ---------------------------------------------------------------------------

export type ProviderCheckTrigger = "startup" | "manual_refresh";

/** 对应 Rust ProviderIssue — 定位到具体 Provider 的问题 */
export interface ProviderIssue {
  /** 问题所属的 Provider */
  provider: ProviderId;
  /** 结构化问题编码 */
  code: string;
  /** 面向展示的问题描述 */
  message: string;
}

/** 对应 Rust ProviderCheckStartedPayload — 生命周期开始事件 */
export interface ProviderCheckStartedPayload {
  /** 本轮检查唯一标识，关联 started/status/completed 事件 */
  run_id: string;
  /** 本轮检查的触发来源 */
  trigger: ProviderCheckTrigger;
}

/** 对应 Rust ProviderStatusPayload — 逐个 Provider 推送的状态事件 */
export interface ProviderStatusPayload {
  /** 本轮检查唯一标识，关联 started/status/completed 事件 */
  run_id: string;
  /** 当前状态所属的 Provider */
  provider: ProviderId;
  /** 已持久化的配置快照 */
  config: ProviderRecord;
  /** 健康检查结果 */
  health: HealthCheckResponse;
  /** 去敏密钥元信息 */
  secret_meta: ProviderSecretMeta;
}

/** 对应 Rust ProviderCheckCompletedPayload — 生命周期正常结束事件 */
export interface ProviderCheckCompletedPayload {
  /** 本轮检查唯一标识，关联 started/status/completed 事件 */
  run_id: string;
  /** 健康检查失败数量 */
  failed: number;
}

/** 对应 Rust ProviderCheckFailedPayload — 生命周期异常终止事件 */
export interface ProviderCheckFailedPayload {
  /** 本轮检查唯一标识，关联 started/status/failed 事件 */
  run_id: string;
  /** 结构化错误码 */
  code: string;
  /** 面向展示或日志的错误信息 */
  message: string;
  /** Provider 级问题列表，仅在可定位到具体 Provider 时存在 */
  issues?: ProviderIssue[];
}
